src-browserify/api/sql.js

```diff
@@ -1,686 +1,697 @@
 var _ = require('underscore');
-var _Promise = require('_Promise');
 var Mustache = require('mustache');
 
-// TODO: jQuery.ajax OR reqwest is expected to be located at global namespace when an SQL object is created
-function SQL(options) {
-  if(cartodb === this || window === this) {
-    return new SQL(options);
-  }
-  if(!options.user) {
-    throw new Error("user should be provided");
-  }
-  var loc = new String(window.location.protocol);
-  loc = loc.slice(0, loc.length - 1);
-  if(loc == 'file') {
-    loc = 'https';
-  }
-
-  this.ajax = options.ajax || (typeof(jQuery) !== 'undefined' ? jQuery.ajax: reqwest);
-  if(!this.ajax) {
-    throw new Error("jQuery or reqwest should be loaded");
-  }
-
-  this.options = _.defaults(options, {
-    version: 'v2',
-    protocol: loc,
-    jsonp: typeof(jQuery) !== 'undefined' ? !jQuery.support.cors: false
-  })
-
-  if (!this.options.sql_api_template) {
-    var opts = this.options;
-    var template = null;
-    if(opts && opts.completeDomain) {
-      template = opts.completeDomain;
-    } else {
-      var host = opts.host || 'cartodb.com';
-      var protocol = opts.protocol || 'https';
-      template = protocol + '://{user}.' + host;
+// NOTE this does not return a SQL directly, but a wrapper, to inject the dependencies
+// e.g. var SQL = require('./sql')({ jQuery: $ /* or reqwest: reqwest */ });
+// @param {Object} _Promise
+// @param {Object} opts hash with keys:
+//   jQuery: {Object} (Optional) has precedence over reqwest if both are present
+//   reqwest: {Object} (Optional)
+module.exports = function(_Promise, opts) {
```

Comment:
Have had a offline conversation with @alonsogarciapablo to clarify the intentions and _why_ I chose to go down this path. While migrating code I have seen the benefits (untangling dependencies and getting a a clear overview) but with more verbose code. Neither of us are completely convinced though, the alternative would be to go back to using window.XXX all other the place. I guess it'll be a tradeoff. @rochoa since you did a similar migration and knows the ins of browserify too, could you have a look too and give an opinion?